DicomUltrasoundPlugin/DicomUltrasoundPlugin.py

```diff
@@ -327,6 +330,78 @@ def examineGeUSMovie(self, filePath):
 
     return [loadable]
 
+  def examineEigenArtemis3DUS(self, filePath):
+    supportedSOPClassUID = '1.2.840.10008.5.1.4.1.1.3.1' # UltrasoundMultiframeImageStorage
+
+    # Quick check of SOP class UID without parsing the file...
+    try:
+      sopClassUID = slicer.dicomDatabase.fileValue(filePath, self.tags['sopClassUID'])
+      if sopClassUID != supportedSOPClassUID:
+        # Unsupported class
+        logging.error("sopClassUID "+sopClassUID+" != supportedSOPClassUID "+supportedSOPClassUID)
+        return []
+    except Exception as e:
+      # Quick check could not be completed (probably Slicer DICOM database is not initialized).
+      # No problem, we'll try to parse the file and check the SOP class UID then.
+      pass
+
+    try:
+      ds = dicom.read_file(filePath, stop_before_pixels=True)
+    except Exception as e:
+      logging.debug("Failed to parse DICOM file: {0}".format(e.msg))
+      return []
+
+    if ds.SOPClassUID != supportedSOPClassUID:
+      # Unsupported class
+      logging.error("sopClassUID "+ds.SOPClassUID+" != supportedSOPClassUID "+supportedSOPClassUID)
+      return []
+
+    if not (ds.Manufacturer == "Eigen" and ds.ManufacturerModelName == "Artemis"):
+      return []
+
+    '''
+    voxelSpacingTag = findPrivateTag(ds, 0x200d, 0x03, "Philips US Imaging DD 036")
```

Comment:
Is this correct?

Reply:
I left it as a placeholder to add code interpreting some of the private attributes. Private attributes contain information that can also be derived from non-private attributes using "private" process, so I just thought I could check if those two result in the same answer as a consistency check. At the time I was working on this, I did not have an example that includes private attributes, but I do have it now, so I will address this.